subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/DependencyCreators.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.plugins.jvm;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.artifacts.ProjectDependency;
+import org.gradle.api.artifacts.dsl.DependencyFactory;
+
+import javax.inject.Inject;
+
+/**
+ * Universal APIs that are available for all {@code dependencies} blocks.
+ *
+ * @since 7.6
+ */
+@Incubating
+public interface DependencyCreators {
```

Comment:
I don't like the inconsistency in the naming of the hierarchy of types here.

We have `Decorator` classes (which aren't GoF Decorators), which extending a `Creators` class (which suggests it is utility methods for a missing `DependencyCreator` type), and which is in turn extended and then implemented by `Dependencies` types.

Why not just call the whole chain `Dependencies`?

`UniversalDependencies` -> `Gradle/Platform/TestFixtures Dependencies` -> `JvmComponentDependencies`.